### ISSUE TITLE
False Positive on spring-security-rsa-1.0.9.RELEASE.jar

### ISSUE THREAD

**Issue by KuSh** [dependency-check/DependencyCheck#3219]
Hi,

False positive on library spring-security-rsa-1.0.9.RELEASE.jar - reported as cpe:/a:vmware:springsource_spring_security
```xml
<dependency>
   <groupId>org.springframework.security</groupId>
   <artifactId>spring-security-rsa</artifactId>
   <version>1.0.9.RELEASE</version>
</dependency>
```

Suppression:
```
<suppress>
   <notes><![CDATA[
   file name: spring-security-rsa-1.0.9.RELEASE.jar
   ]]></notes>
   <packageUrl regex="true">^pkg:maven/org\.springframework\.security/spring\-security\-rsa@.*$</packageUrl>
   <cpe>cpe:/a:vmware:springsource_spring_security</cpe>
</suppress>
```

Regards,
Nicolas

**Comment by KuSh** [comment]
So suppression could be a bit "large" according to #3218 .. a better one could be :

```
<suppress>
    <notes><![CDATA[
    file name: spring-security-rsa-1.0.9.RELEASE.jar
    ]]></notes>
    <packageUrl regex="true">^pkg:maven/org\.springframework\.security/spring\-security\-rsa@.*$</packageUrl>
    <cve>CVE-2011-2731</cve>
    <cve>CVE-2011-2732</cve>
    <cve>CVE-2012-5055</cve>
</suppress>
```

**Comment by aikebah** [comment]
Couldn't find any references of the spring-security-rsa module apart from [Dave Syers repository](https://github.com/dsyer/spring-security-rsa). Looks as if despite the maven coordinates it's not (no longer? not yet?) part of the official spring libraries, so I tend to think that your original suppression proposal would be fitting in this case.

It references `http://github.com/spring-projects/spring-security-rsa` as the SCM-location which is non-existent.
It references `http://github.com/spring-projects/spring-security-oauth` as the project URL, which also holds no traces of spring-security-rsa that I could spot.

@dsyer Would you be able to shed some light on this? Is spring-security-rsa indeed a standalone library outside of spring itself as it appears to be? Wouldn't want to add a suppression for the CPE in error, but from what I could surface it appears to be the right thing to do for this library.

**Comment by dsyer** [comment]
I'm still hosting it, but it's used in a couple of places in Spring Cloud, and possibly elsewhere, so I guess that makes it "official". Originally I think it was used in the OAuth module, but that one is now in maintenance mode.

I bumped the BC dependencies and released 1.0.10 today, but if you think 1.0.9 is also clear (I believe so based on my understanding), then please suppress the error.

**Comment by aikebah** [comment]
@dsyer Thanks, that clears the confusion for me. Suppressing its detection as a library of spring-security then is the correct approach (as only the composing parts of spring-security would have their vulnerabilities classified by NIST NVD as belonging to the spring-security CPE)
